tests/histogram.rs

```diff
@@ -83,6 +83,15 @@ fn record_past_trackable_max() {
     assert!(h.record(3 * TRACKABLE_MAX).is_err());
 }
 
+#[test]
+fn saturating_record_past_trackable_max() {
+    let mut h = Histogram::<u64>::new_with_max(TRACKABLE_MAX, SIGFIG).unwrap();
+    h.saturating_record(3 * TRACKABLE_MAX);
```

Comment:
I'd feel better if this also had asserts for clamping on the low end, and a value in the middle that wasn't clamped.

Reply:
Good point!

However, I'm running into issues. The following test fails:
```rust
let mut h = Histogram::<u64>::new_with_bounds(1000, TRACKABLE_MAX, SIGFIG).unwrap();
h.saturating_record(1); // clamped below
h.saturating_record(1000*1000); // not clamped
h.saturating_record(3 * TRACKABLE_MAX); // clamped above
assert_eq!(h.count_at(h.low()), 1);
```
`h.low()` correctly returns 1000, but `iter_recorded` gives a first sample of
```rust
IterationValue { value_iterated_to: 511, quantile: 0.3333333333333333, quantile_iterated_to: 0.3333333333333333, count_at_value: 1, count_since_last_iteration: 1 }
```
This suggests that there are bins below `h.low()`, which isn't entirely surprising if you know the internal representation of HdrHistogram, but will likely be unexpected for many users. I wonder how we might go about fixing that...

Reply:
The way to fix this for now is to create the histogram with a min bound of 512, and then check that a recorded value of `1` actually adds to the count at 511, but I'd prefer being able to query the 511 from the histogram. I think this suggests to me that `low()` currently returns the wrong value. It should return the *actual* low, not the one passed in by the user.

Reply:
I think you're right -- you configure the histogram with what you NEED to be able to represent, and it will do that, but may also support other values. That semantic makes sense to me, so we might as well expose the actual limits.

I wonder if we should record the actual limits rather than the requested range when serializing? That's arguably more useful when determining compatibility for, say, deserializing a histogram *into* another histogram...